chat-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -203,24 +237,56 @@ export async function getContentForText({
     text,
     userIp: ipAddress,
   });
-  return store.findNearestNeighbors(embedding, findNearestNeighborsOptions);
+  return await store.findNearestNeighbors(
+    embedding,
+    findNearestNeighborsOptions
+  );
+}
+
+interface AddMessagesToDatabaseParams {
+  conversationId: ObjectId;
+  userMessageContent: string;
+  assistantMessageContent: string;
+  conversations: ConversationsServiceInterface;
+}
+export async function addMessagesToDatabase({
+  conversationId,
+  userMessageContent,
+  assistantMessageContent,
+  conversations,
+}: AddMessagesToDatabaseParams) {
+  // TODO: consider refactoring addConversationMessage to take in an array of messages.
+  // Would limit database calls.
+  const userMessage = await conversations.addConversationMessage({
+    conversationId,
+    content: userMessageContent,
+    role: "user",
+  });
+  const assistantMessage = await conversations.addConversationMessage({
+    conversationId,
+    content: assistantMessageContent,
+    role: "assistant",
+  });
+  return { userMessage, assistantMessage };
 }
 
 export interface GenerateFurtherReadingParams {
   chunks: EmbeddedContent[];
+  hasHeading?: boolean;
 }
 export function generateFurtherReading({
   chunks,
+  hasHeading = true,
 }: GenerateFurtherReadingParams) {
   if (chunks.length === 0) {
     return "";
   }
-  const heading = "\n\nArticles Referenced:\n";
+  const heading = hasHeading ? "\n\nFurther Reading:\n" : "";
```

Comment:
```suggestion
  const heading = hasHeading ? "\n\nFurther Reading:\n\n" : "";
```

Markdown renders newlines as spaces in this case - we need a double newline to indicate the start of a new paragraph and actually move the first link to the next line.

Reply:
ah, of course! oops will adjust